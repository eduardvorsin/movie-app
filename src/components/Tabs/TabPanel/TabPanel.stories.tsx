import type { Meta, StoryObj } from '@storybook/react';
import TabPanel from './TabPanel';

const meta: Meta<typeof TabPanel> = {
	title: 'components/Tabs/TabPanel',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for TabPanel'
		},
		testId: {
			description: 'Id for testing the component',
		},
		children: {
			description: 'Contents of the TabPanel component',
		},
		label: {
			description: 'Text to display in TabButton',
		},
	},
	component: TabPanel,
	parameters: {
		docs: {
			description: {
				component: 'The TabPanel component is a container used in the Tabs component to display the contents of a specific tab or section on a web page. It allows the user to see and interact with the content associated with the selected tab, providing ease of navigation and accessibility of the interface.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof TabPanel>;

export const Default: Story = {
	args: {
		label: 'tab-label',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Officia quia illum eligendi adipisci magni, atque animi facere odit. Ex in illo obcaecati natus culpa consequuntur nobis tempore quidem rem eum.Dicta deserunt vero consequatur veritatis veniam tenetur quibusdam vitae aspernatur perspiciatis quas modi voluptatum voluptatem impedit optio soluta eum, explicabo sequi totam quisquam doloribus consectetur illum tempora. Reiciendis, accusamus corporis.',
	},
	parameters: {
		docs: {
			description: {
				story: 'Default state for TabButton',
			},
		},
	},
};
