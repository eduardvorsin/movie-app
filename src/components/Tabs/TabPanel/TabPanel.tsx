import { ReactNode } from 'react';

export type Props = {
	label: string,
	className?: string,
	testId?: string,
	children: ReactNode,
};

export default function TabPanel({
	label,
	className,
	testId,
	children,
}: Props) {
	return (
		<div
			className={`w-full ${className}`}
			data-label={label}
			data-testid={testId}
		>
			{children}
		</div>

	);
}