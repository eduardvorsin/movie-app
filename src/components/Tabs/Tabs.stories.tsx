import type { Meta, StoryObj } from '@storybook/react';
import Tabs from './Tabs';
import TabPanel from './TabPanel/TabPanel';

const meta: Meta<typeof Tabs> = {
	title: 'UI/Tabs',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for Tabs'
		},
		children: {
			control: false,
			description: 'Contents of tabs',
		},
		testId: {
			description: 'Id for testing the component',
		},
		onSelect: {
			control: false,
			description: 'Id for testing the component',
		},
		defaultSelected: {
			description: 'The tab that will be displayed by default',
		},
		isDisabled: {
			description: 'Allows you to make buttons for tabs inactive',
		},
		fitted: {
			description: 'Allows tab buttons to take up the entire allowable width',
		}
	},
	component: Tabs,
	parameters: {
		docs: {
			description: {
				component: 'The "Tabs" component is an element that is used to organize content into several tabs, allowing the user to choose between different sections of information without reloading the page. It improves navigation and accessibility by allowing users to easily switch between related sections of content.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Tabs>;

const tabLabels = ['banana', 'orange', 'apple', 'avocado', 'blackberry', 'blueberry', 'cherry', 'cocount', 'grape', 'lemon', 'lime', 'nectarine', 'watermelon', 'raspberry', 'strawberry', 'mango', 'pear', 'papaya', 'pomegranate', 'peach', 'pineapple', 'cabbage', 'broccoli', 'mushroom', 'zucchini', 'red chilli pepper', 'sweet potato'];
const tabPanels = tabLabels.map((label, index) => (
	<TabPanel
		className='p-2'
		key={index}
		label={label}
	>
		{label}
	</TabPanel>
));

export const Default: Story = {
	args: {
		id: 'tabs',
		children: tabPanels.slice(0, 3),
	},
	parameters: {
		docs: {
			description: {
				story: 'Default state of tabs',
			},
		},
	},
};

export const Disabled: Story = {
	args: {
		id: 'tabs',
		children: tabPanels.slice(0, 3),
		isDisabled: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Makes tab buttons inactive. It is useful in cases when you need to limit the user\'s ability from repeated clicks, or just make it temporarily inactive',
			},
		},
	},
};

export const Fitted: Story = {
	args: {
		id: 'tabs',
		children: tabPanels.slice(0, 3),
		fitted: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Tab buttons occupy the entire free width, in equal parts',
			},
		},
	},
};

export const ManyTabs: Story = {
	args: {
		id: 'tabs',
		children: tabPanels,
	},
	parameters: {
		docs: {
			description: {
				story: 'Tab buttons occupy the entire free width, in equal parts',
			},
		},
	},
};

export const EventCallbacks: Story = {
	argTypes: {
		onSelect: {
			action: 'Selected',
		},
	},
	args: {
		id: 'tabs',
		children: tabPanels.slice(0, 3),
	},
	parameters: {
		docs: {
			description: {
				story: 'The callback function is called when another tab is selected',
			},
		},
	},
};

