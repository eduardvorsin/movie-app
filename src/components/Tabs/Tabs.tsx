import { KeyboardEventHandler, MouseEventHandler, ReactElement, useRef, useState } from 'react';
import TabButton from './TabButton/TabButton';
import { Props as TabPanelProps } from './TabPanel/TabPanel';

type Props = {
	children: ReactElement<TabPanelProps>[],
	id: string,
	onSelect: (index: number) => void,
	defaultSelected?: number,
	className?: string,
	testId?: string,
	isDisabled?: boolean,
	fitted?: boolean,
};

export default function Tabs({
	children,
	id,
	onSelect,
	defaultSelected = 0,
	className,
	testId,
	isDisabled,
	fitted,
}: Props) {
	const tabsGroupRef = useRef<HTMLDivElement>(null);
	const [selectedTabIndex, setSelectedTabIndex] = useState<number>(defaultSelected);

	const clickHandler: MouseEventHandler<HTMLButtonElement> = (e) => {
		const { index } = e.currentTarget.dataset;
		if (!index) return;
		setSelectedTabIndex(+index);
		onSelect(+index);
	}

	const keyDownHandler: KeyboardEventHandler<HTMLButtonElement> = (e) => {
		const keys = ['ArrowLeft', 'ArrowRight', 'Home', 'End'];
		const { nextElementSibling, previousElementSibling } = e.currentTarget;
		if (!keys.includes(e.code)) return;

		let nextIndex = -1;
		if (e.code === 'ArrowLeft') {
			nextIndex = previousElementSibling ? selectedTabIndex - 1 : children.length - 1;
		} else if (e.code === 'ArrowRight') {
			nextIndex = nextElementSibling ? selectedTabIndex + 1 : 0;
		} else if (e.code === 'Home') {
			nextIndex = 0;
		} else if (e.code === 'End') {
			nextIndex = children.length - 1;
		}

		(tabsGroupRef.current?.children[nextIndex] as HTMLButtonElement).focus();
		setSelectedTabIndex(nextIndex);
	}

	const classes = [
		'w-full bg-neutral-100 dark:bg-dark-neutral-200 rounded-[0.1875rem] border-2 border-neutral-300 dark:border-dark-neutral-350',
		className,
	].join(' ');

	const tabButtonClasses = [
		'mr-1 last:mr-0 ',
		fitted ? 'grow' : '',
	].join(' ');

	return (
		<div
			className={classes}
			id={id}
			data-testid={testId}
		>
			<div
				role='tablist'
				className='flex p-2 overflow-hidden'
				ref={tabsGroupRef}
			>
				{children.map((child, index) => (
					<TabButton
						className={tabButtonClasses}
						key={child.props.label}
						label={child.props.label}
						id={`${id}-tab-${index}`}
						onClick={clickHandler}
						onKeyDown={keyDownHandler}
						index={index}
						isActive={index === selectedTabIndex}
						ariaControls={`${id}-tabpanel-${index}`}
						isDisabled={isDisabled}
					/>
				))}
			</div>
			<div
				className='flex min-h-[5rem] text-neutral-1000 dark:text-dark-neutral-900 p-2 rounded-b-[0.1875rem] border-t-2 border-neutral-300 dark:border-dark-neutral-350'
				id={`${id}-tabpanel-${selectedTabIndex}`}
				role='tabpanel'
				aria-labelledby={`${id}-tab-${selectedTabIndex}`}
				tabIndex={0}
			>
				{children[selectedTabIndex]}
			</div>
		</div>
	);
};
