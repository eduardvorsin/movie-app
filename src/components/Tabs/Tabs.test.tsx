import { render, screen } from '@testing-library/react';
import Tabs from './Tabs';
import TabPanel from './TabPanel/TabPanel';
import userEvent from '@testing-library/user-event';

describe('Tabs tests', () => {
	it('is rendered correctly', () => {
		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-tabs')).toBeInTheDocument();
	});

	it('when you click on the tab button, the mock function is triggered', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<Tabs
				id='tabs'
				onSelect={mockFn}
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('tab', { name: /orange/i }));

		expect(mockFn).toHaveBeenCalledTimes(1);
	});

	it('if isDisabled is true then the tab buttons become inactive', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<Tabs
				id='tabs'
				onSelect={mockFn}
				isDisabled
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('tab', { name: /orange/i }));

		expect(mockFn).toHaveBeenCalledTimes(0);
	});

	it('switch to the next tab when arrow right is pressed', async () => {
		const user = userEvent.setup();

		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		await user.tab();
		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /apple/i })).toHaveFocus();

		await user.keyboard('[ArrowRight]');

		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /banana/i })).toHaveFocus();
	});

	it('switch to the previous tab when arrow left is pressed', async () => {
		const user = userEvent.setup();

		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		await user.tab();
		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /apple/i })).toHaveFocus();

		await user.keyboard('[ArrowLeft]');

		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /orange/i })).toHaveFocus();
	});

	it('switch to the first tab when home is pressed', async () => {
		const user = userEvent.setup();

		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
				defaultSelected={2}
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		await user.tab();
		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /orange/i })).toHaveFocus();

		await user.keyboard('[Home]');

		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /apple/i })).toHaveFocus();
	});

	it('switch to the last tab when end is pressed', async () => {
		const user = userEvent.setup();

		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		await user.tab();
		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /apple/i })).toHaveFocus();

		await user.keyboard('[End]');

		expect(screen.getByRole<HTMLButtonElement>('tab', { name: /orange/i })).toHaveFocus();
	});

	it('is a basic shapshot', () => {
		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-tabs')).toMatchSnapshot();
	});

	it('is a snapshot with isDisabled set to true', () => {
		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
				isDisabled
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-tabs')).toMatchSnapshot();
	});

	it('is a snapshot with fitted set to true', () => {
		render(
			<Tabs
				id='tabs'
				onSelect={() => { }}
				testId='test-tabs'
				fitted
			>
				<TabPanel label='apple'>apple</TabPanel>
				<TabPanel label='banana'>banana</TabPanel>
				<TabPanel label='orange'>orange</TabPanel>
			</Tabs>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-tabs')).toMatchSnapshot();
	});
});