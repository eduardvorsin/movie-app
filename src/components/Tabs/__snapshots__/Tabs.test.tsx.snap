// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs tests is a basic shapshot 1`] = `
<div
  class="w-full bg-neutral-100 dark:bg-dark-neutral-200 rounded-[0.1875rem] border-2 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150 "
  data-testid="test-tabs"
  id="tabs"
>
  <div
    class="transition-colors duration-150 "
  >
    <div
      class="flex p-2 overflow-x-auto no-scrollbar "
      role="tablist"
    >
      <button
        aria-controls="tabs-tabpanel-0"
        aria-selected="true"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-blue-700 dark:bg-blue-300 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:enabled:hover:bg-blue-200 dark:enabled:active:bg-blue-100  shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow-0"
        data-index="0"
        id="tabs-tab-0"
        role="tab"
        tabindex="0"
      >
        apple
      </button>
      <button
        aria-controls="tabs-tabpanel-1"
        aria-selected="false"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 enabled:hover:bg-neutral-800 enabled:active:bg-neutral-900 dark:enabled:hover:bg-dark-neutral-800 dark:enabled:active:bg-dark-neutral-900  shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow-0"
        data-index="1"
        id="tabs-tab-1"
        role="tab"
        tabindex="-1"
      >
        banana
      </button>
      <button
        aria-controls="tabs-tabpanel-2"
        aria-selected="false"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 enabled:hover:bg-neutral-800 enabled:active:bg-neutral-900 dark:enabled:hover:bg-dark-neutral-800 dark:enabled:active:bg-dark-neutral-900  shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow-0"
        data-index="2"
        id="tabs-tab-2"
        role="tab"
        tabindex="-1"
      >
        orange
      </button>
    </div>
  </div>
  <div
    aria-labelledby="tabs-tab-0"
    class="flex min-h-[4rem] text-neutral-1000 dark:text-dark-neutral-900 rounded-b-[0.1875rem] border-t-2 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150"
    id="tabs-tabpanel-0"
    role="tabpanel"
    tabindex="0"
  >
    <div
      class="w-full undefined"
      data-label="apple"
    >
      apple
    </div>
  </div>
</div>
`;

exports[`Tabs tests is a snapshot with fitted set to true 1`] = `
<div
  class="w-full bg-neutral-100 dark:bg-dark-neutral-200 rounded-[0.1875rem] border-2 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150 "
  data-testid="test-tabs"
  id="tabs"
>
  <div
    class="transition-colors duration-150 "
  >
    <div
      class="flex p-2 overflow-x-auto no-scrollbar "
      role="tablist"
    >
      <button
        aria-controls="tabs-tabpanel-0"
        aria-selected="true"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-blue-700 dark:bg-blue-300 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:enabled:hover:bg-blue-200 dark:enabled:active:bg-blue-100  shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow"
        data-index="0"
        id="tabs-tab-0"
        role="tab"
        tabindex="0"
      >
        apple
      </button>
      <button
        aria-controls="tabs-tabpanel-1"
        aria-selected="false"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 enabled:hover:bg-neutral-800 enabled:active:bg-neutral-900 dark:enabled:hover:bg-dark-neutral-800 dark:enabled:active:bg-dark-neutral-900  shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow"
        data-index="1"
        id="tabs-tab-1"
        role="tab"
        tabindex="-1"
      >
        banana
      </button>
      <button
        aria-controls="tabs-tabpanel-2"
        aria-selected="false"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 enabled:hover:bg-neutral-800 enabled:active:bg-neutral-900 dark:enabled:hover:bg-dark-neutral-800 dark:enabled:active:bg-dark-neutral-900  shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow"
        data-index="2"
        id="tabs-tab-2"
        role="tab"
        tabindex="-1"
      >
        orange
      </button>
    </div>
  </div>
  <div
    aria-labelledby="tabs-tab-0"
    class="flex min-h-[4rem] text-neutral-1000 dark:text-dark-neutral-900 rounded-b-[0.1875rem] border-t-2 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150"
    id="tabs-tabpanel-0"
    role="tabpanel"
    tabindex="0"
  >
    <div
      class="w-full undefined"
      data-label="apple"
    >
      apple
    </div>
  </div>
</div>
`;

exports[`Tabs tests is a snapshot with isDisabled set to true 1`] = `
<div
  class="w-full bg-neutral-100 dark:bg-dark-neutral-200 rounded-[0.1875rem] border-2 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150 "
  data-testid="test-tabs"
  id="tabs"
>
  <div
    class="transition-colors duration-150 "
  >
    <div
      class="flex p-2 overflow-x-auto no-scrollbar snap-mandatory snap-x"
      role="tablist"
    >
      <button
        aria-controls="tabs-tabpanel-0"
        aria-selected="true"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-blue-700 dark:bg-blue-300 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:enabled:hover:bg-blue-200 dark:enabled:active:bg-blue-100 opacity-disabled cursor-not-allowed shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow-0"
        data-index="0"
        disabled=""
        id="tabs-tab-0"
        role="tab"
        tabindex="0"
      >
        apple
      </button>
      <button
        aria-controls="tabs-tabpanel-1"
        aria-selected="false"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 enabled:hover:bg-neutral-800 enabled:active:bg-neutral-900 dark:enabled:hover:bg-dark-neutral-800 dark:enabled:active:bg-dark-neutral-900 opacity-disabled cursor-not-allowed shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow-0"
        data-index="1"
        disabled=""
        id="tabs-tab-1"
        role="tab"
        tabindex="-1"
      >
        banana
      </button>
      <button
        aria-controls="tabs-tabpanel-2"
        aria-selected="false"
        class="text-100 font-medium p-1 rounded-[0.1875rem] text-center text-neutral-100 dark:text-dark-neutral-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 enabled:hover:bg-neutral-800 enabled:active:bg-neutral-900 dark:enabled:hover:bg-dark-neutral-800 dark:enabled:active:bg-dark-neutral-900 opacity-disabled cursor-not-allowed shrink-0 mr-1 last:mr-0 enabled:snap-start enabled:snap-always enabled:scroll-mx-2 break-all grow-0"
        data-index="2"
        disabled=""
        id="tabs-tab-2"
        role="tab"
        tabindex="-1"
      >
        orange
      </button>
    </div>
  </div>
  <div
    aria-labelledby="tabs-tab-0"
    class="flex min-h-[4rem] text-neutral-1000 dark:text-dark-neutral-900 rounded-b-[0.1875rem] border-t-2 border-neutral-300 dark:border-dark-neutral-350 transition-colors duration-150"
    id="tabs-tabpanel-0"
    role="tabpanel"
    tabindex="0"
  >
    <div
      class="w-full undefined"
      data-label="apple"
    >
      apple
    </div>
  </div>
</div>
`;
